### PR TITLE
Use the cythonized DoFnContext everywhere.

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Of particular note we want this to be strongly typed in
the DoFnRunner as one interacts with it for every element
processed.

Also moves the creation of this context, which is an implementation
detail, from the runner to the API for future flexibility.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @aaltay

File: sdks/python/apache_beam/runners/common.py [hunk 1]
```diff
@@ -59,12 +59,16 @@ def __init__(self,
...
                windowing,
-               context,
-               tagged_receivers,
+               context=None,
+               tagged_receivers=None,
                logger=None,
                step_name=None,
                # Preferred alternative to logger
-               logging_context=None):
+               # TODO(robertwb): Remove once all runners are updated.
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/common.py]: What will be removed when all runners are updated? logging_context and state? Or context parameter? Or the None defaults?

**Comment by aaltay** [comment]
LGTM, Thank you Robert.

**Comment by aaltay** [comment]
R: @dhalperi I believe this is ready to merge. Please review.

**Comment by dhalperi** [comment]
LGTM

**Pull request closed without merging**